content/guides/code-splitting.md

```diff
@@ -1,41 +1,174 @@
 
-Learn [how to split on demand](/guides/code-splitting-async) using `import()` or `require.ensure()`.
+There are multiple plugins and loaders in webpack's ecosystem that can help with splitting and managing split code. The most widely utilized of these is the [`CommonsChunkPlugin`](/plugins/commons-chunk-plugin), a fairly advanced tool that allows us to extract common dependencies into an existing entry chunk or an entirely new chunk. Let's use this to de-duplicate the `moment` dependency from the previous example:
+
+__webpack.config.js__
+
+``` js
+const path = require('path');
+const webpack = require('webpack');
+
+module.exports = {
+  entry: {
+    index: './src/index.js',
+    vendor: [
+      'lodash'
+    ]
+  },
+  plugins: [
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'vendor' // Specify the common bundle's name.
+    })
+  ],
+  output: {
+    filename: '[name].bundle.js',
+    path: path.resolve(__dirname, 'dist')
+  }
+};
+```
+
+This will yield the following build result (notice how our `index.bundle.js` has shrunk in size):
+
+?> Add bash example of webpack output
+
+Here are some other useful plugins and loaders provide by the community for splitting code:
+
+- [`ExtractTextPlugin`](/plugins/extract-text-webpack-plugin): Useful for splitting CSS out from the main application.
+- [`bundle-loader`](/loaders/bundle-loader): Used to split code and lazy load the resulting bundles.
+- [`promise-loader`](https://github.com/gaearon/promise-loader): Similar to the `bundle-loader` but uses promises.
+
+
+## Dynamic Imports
+
+Two similar techniques are supported by webpack when it comes to dynamic code splitting. The first and more preferable approach is use to the [`import()` syntax](/api/module-methods#import-) that conforms to the [ECMAScript proposal](https://github.com/tc39/proposal-dynamic-import) for dynamic imports. The legacy, webpack-specific approach is to use [`require.ensure`](/api/module-methods#require-ensure). Let's try using the first of these two approaches...
+
+Here, instead of statically importing lodash, we'll use dynamic importing to separate it into a separate chunk:
+
+__src/index.js__
+
+``` diff
+- import _ from 'lodash';
+-
+- function component() {
++ function getComponent() {
+-   var element = document.createElement('div');
+-
+-   // Lodash, now imported by this script
+-   element.innerHTML = _.join(['Hello', 'webpack'], ' ');
++   return import(/* webpackChunkName: "lodash" */ 'lodash').then(module => {
++     var element = document.createElement('div');
++
++     element.innerHTML = _.join(['Hello', 'webpack'], ' ');
++
++     return element;
++
++   }).catch(error => 'An error occurred while loading the component');
+  }
+
+- document.body.appendChild(component());
++ getComponent().then(component => {
++   document.body.appendChild(component);
++ })
+```
+
+Now, when we run webpack, we should see lodash separated out to a separate bundle:
+
+?> Add bash example of webpack output
```

Comment:
Nope this results in `index.bundle.js` and `vendor.bundle.js`, not `lodash.bundle.js`.

Ahhh, this is a webpack 2.4.0+ specific thing, I still run 2.6.1, so that's probably why.

We should probably make a note of which version to use for the guides. I'm assuming the latest stable build for the major version?